### PR TITLE
Fix two undefined variable errors in numpy.ma that caused two different bugs

### PR DESCRIPTION
**Pull request by astrofrog**
For the unicode issue, we could also simply map `unicode = str` on Python 3+, which has the benefit of making static code analysis tools happier.

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def default_fill_value(obj):
         defval = default_filler['i']
     elif isinstance(obj, str):
         defval = default_filler['S']
-    elif isinstance(obj, unicode):
+    elif sys.version_info[0] < 3 and isinstance(obj, unicode):
```
Comment (charris) [inline numpy/ma/core.py]: Might be better to import `unicode` and `bytes` from `numpy.compat` and use those instead, that way the two string types will be orthogonal for both Python 2 and Python 3.

Comment (astrofrog) [inline numpy/ma/core.py]: @charris - just to check, are you suggesting changing line 220 to have bytes?

Comment (charris) [inline numpy/ma/core.py]: Yes. If you look in `numpy/compat/py3k.py` you will see that both unicode and bytes are defined so as to cover the correct types in both python 2 and python 3.

Comment (charris) [inline numpy/ma/core.py]: Maybe we should use compatible filler values also, `sixu('N/A')` for unicode and `b'N/A'` for bytes. Not sure about the usefulness of 'N/A' however.

**Comment by astrofrog** [comment]
@charris - I made the changes you requested, but note that this causes a test to fail. However, before I fix the test I want to make sure I understand it. The test lines that fail are:

```python
        fval = _check_fill_value(None, "|S3")
        assert_equal(fval, default_fill_value("|S3"))
```

the issue is that ``default_fill_value("|S3")`` now returns ``array('N/A', dtype='<U3')`` - however I wonder if there was a typo in the original test, since ``default_fill_value`` appears to be called with what looks like a dtype even though the content is irrelevant, and instead ``default_fill_value`` only cares about the fact that what is passed in is a (unicode) string. What do you think?

**Comment by charris** [comment]
I think my head hurts ;) But I think you are correct that the test is not right.  `"|S3"` will be bytes in python 2 and unicode in python 3, which I assume is what you are seeing. It isn't clean to me how the string filler is used in any case, but that is another problem...

**Comment by astrofrog** [comment]
@charris - would you recommend simply removing those two lines from the test in that case? Or do you have suggestions for what it should be instead?

**Comment by charris** [comment]
I'd just replace `default_fill_value("|S3")` by `default_fill_value(b"|S3")`. However, using `"|S3"` as both type and string instance is confusing, maybe using `"astring"` or some such would be better.

**Comment by astrofrog** [comment]
@charris - ok, done!

**Comment by charris** [comment]
Thanks Thomas.

**Pull request merged**